packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -189,6 +216,21 @@ export function updateInput(
       checkAttributeStringCoercion(name, 'name');
     }
     node.name = toString(getToStringValue(name));
+
+    if (needUpdateRadioGroupTrackedValue) {
+      const group = findRadioGroup(node);
```

Comment:
This is going to be a fragile fix due to it querying the DOM during the mutation where things can happen before or after depending on tree order. It would also potentially trigger thrashing of the DOM internal caches instead of all writes and then reads.

If this worked before I think the solution is probably more likely something about the timing of when the mutation should happen to replicate the previous order of operations.